### PR TITLE
Add block number parameter to the KM PTT request

### PR DESCRIPTION
**Pull request by lenak25**

### CODE REVIEW

**Comment by lenak25** [comment]
@AvishaiW ready for your review

**Comment by lenak25** [comment]
@AvishaiW ready for review once again :)

File: test/principal_test.js [hunk 1]
```diff
@@ -5,20 +5,22 @@ const ControllerBuilder = require('./testUtils/quickBuilderUtil');
...
 const TEST_TREE = require('./test_tree').TEST_TREE;
 
-
 let receivedRequest = false;
+let response = null;
+let fakeAddresses = null;
+let fakeBlockNumber = null;
 
-it('#1 Should Recieve Encrypted response message from mock principal', async function() {
+it('#1 Should Receive Encrypted response message from mock principal', async function() {
```
Comment (AvishaiW) [inline test/principal_test.js]: Excellent!

File: test/principal_test.js [hunk 2]
```diff
@@ -56,17 +58,23 @@ it('#2 Should Simulate the principal node and run GetStateKeysAction', async fun
...
+          addresses: fakeAddresses,
+          blockNumber: fakeBlockNumber
+        },
     );
     await testUtils.sleep(1500);
     await mainController.shutdownSystem();
     controllers.coreServer.disconnect();
     principalMock.destroy(server);
-    assert(receivedRequest, 'The principal mock never recived a message');
+    assert(receivedRequest, 'The principal mock never received a message');
```
Comment (AvishaiW) [inline test/principal_test.js]: Very consistent!

File: test/verifier_test.js [hunk 3]
```diff
@@ -634,16 +634,20 @@ describe('Verifier tests', function() {
 
     return new Promise(async function(resolve) {
       let a = await initStuffForTaskCreation();
-      let task = new DeployTask(a.secretContractAddress, a.preCode, a.encryptedArgs, a.encryptedFn, a.userDHKey, a.gasLimit, a.secretContractAddress);
+      let blockNumber = a.expectedParams.firstBlockNumber + 50;
+
+      let task = new DeployTask(a.secretContractAddress, a.preCode, a.encryptedArgs, a.encryptedFn, a.userDHKey,
+        a.gasLimit, a.secretContractAddress, 0);
```
Comment (AvishaiW) [inline test/verifier_test.js]: It might be clearer if you put a const `BLOCK_NUMBER=0` or something in that sort. It seems to come up a lot in this file.

Comment (lacabra) [reply test/verifier_test.js]: Where does the `+50` come from?

Comment (lenak25) [reply test/verifier_test.js]: The [EnigmaContractMock](https://github.com/enigmampc/enigma-p2p/blob/master/test/ethereum/EnigmaContractMock.js) used in the test is being initialized with epochs in the size of [100](https://github.com/enigmampc/enigma-p2p/blob/master/test/ethereum/utils.js#L93) blocks so the number 50 represents the middle of an epoch.

**Pull request merged**